// @aps_sdk/model-derivative/dist/custom-code/ModelDerivativeClient.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelDerivativeClient = void 0;
const api_1 = require("../api");
const model_1 = require("../model");
class ModelDerivativeClient {
    constructor(sdkManager) {
        this.derivativesApi = new api_1.DerivativesApi(sdkManager);
        this.informationalApi = new api_1.InformationalApi(sdkManager);
        this.jobsApi = new api_1.JobsApi(sdkManager);
        this.manifestApi = new api_1.ManifestApi(sdkManager);
        this.metadataApi = new api_1.MetadataApi(sdkManager);
        this.thumbnailsApi = new api_1.ThumbnailsApi(sdkManager);
    }
    //#region InformationalAPi
    getFormats(accessToken, optionalArgs) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.informationalApi.getFormats(accessToken, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.ifModifiedSince, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.acceptEncoding, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.options);
            return response.content;
        });
    }
    //#endregion InformationalApi
    //#region JobsApi
    specifyReferences(accessToken, urn, specifyReferencesRequest, optionalArgs) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.jobsApi.specifyReferences(accessToken, urn, (_a = optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.region) !== null && _a !== void 0 ? _a : model_1.Region.Us, specifyReferencesRequest, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.options);
            return response.content;
        });
    }
    startJob(accessToken, jobPayload, optionalArgs) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.jobsApi.startJob(accessToken, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.xAdsForce, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.xAdsDerivativeFormat, (_a = optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.region) !== null && _a !== void 0 ? _a : model_1.Region.Us, jobPayload, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.options);
            return response.content;
        });
    }
    //#endregion JobsApi
    //#region ManifestApi
    getManifest(accessToken, urn, optionalArgs) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.manifestApi.getManifest(accessToken, urn, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.acceptEncoding, (_a = optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.region) !== null && _a !== void 0 ? _a : model_1.Region.Us, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.options);
            return response.content;
        });
    }
    deleteManifest(accessToken, urn, optionalArgs) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.manifestApi.deleteManifest(accessToken, urn, (_a = optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.region) !== null && _a !== void 0 ? _a : model_1.Region.Us, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.options);
            return response.content;
        });
    }
    //#endregion ManifestApi
    //#region DerivativesApi
    getDerivativeUrl(accessToken, derivativeUrn, urn, optionalArgs) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.derivativesApi.getDerivativeUrl(accessToken, derivativeUrn, urn, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.minutesExpiration, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.responseContentDisposition, (_a = optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.region) !== null && _a !== void 0 ? _a : model_1.Region.Us, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.options);
            if (response.response.headers) {
                var cookies = response.response.headers.get("Set-Cookie").flatMap((a) => a.slice(a.indexOf('=') + 1, a.indexOf(';')));
                const url = new URL(response.content.url);
                url.searchParams.set("Policy", cookies[0]);
                url.searchParams.set("Key-Pair-Id", cookies[1]);
                url.searchParams.set("Signature", cookies[2]);
                response.content.url = url.href;
            }
            return response.content;
        });
    }
    headCheckDerivative(accessToken, urn, derivativeUrn, optionalArgs) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.derivativesApi.headCheckDerivative(accessToken, urn, derivativeUrn, (_a = optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.region) !== null && _a !== void 0 ? _a : model_1.Region.Us, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.options);
            if (response.response.status == 202) {
                return (new model_1.DerivativeHead(true));
            }
            else {
                response.content.isProcessing = false;
                return response.content;
            }
        });
    }
    //#endregion DerivativesApi
    //#region thumbnailsApi
    getThumbnail(accessToken, urn, optionalArgs) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.thumbnailsApi.getThumbnail(accessToken, urn, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.width, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.height, (_a = optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.region) !== null && _a !== void 0 ? _a : model_1.Region.Us, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.options);
            return response.content;
        });
    }
    //#endregion thumbnailsApi
    //#region metadataApi
    getObjectTree(accessToken, urn, modelGuid, optionalArgs) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.metadataApi.getObjectTree(accessToken, urn, modelGuid, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.acceptEncoding, (_a = optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.region) !== null && _a !== void 0 ? _a : model_1.Region.Us, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.xAdsForce, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.xAdsDerivativeFormat, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.forceget, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.objectId, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.level, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.options);
            if (response.response.status == 202) {
                return (new model_1.ObjectTree(true));
            }
            else {
                response.content.isProcessing = false;
                return response.content;
            }
        });
    }
    getModelViews(accessToken, urn, optionalArgs) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.metadataApi.getModelViews(accessToken, urn, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.acceptEncoding, (_a = optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.region) !== null && _a !== void 0 ? _a : model_1.Region.Us, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.options);
            return response.content;
        });
    }
    getAllProperties(accessToken, urn, modelGuid, optionalArgs) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.metadataApi.getAllProperties(accessToken, urn, modelGuid, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.acceptEncoding, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.xAdsForce, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.xAdsDerivativeFormat, (_a = optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.region) !== null && _a !== void 0 ? _a : model_1.Region.Us, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.objectId, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.forceget, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.options);
            if (response.response.status == 202) {
                return (new model_1.AllProperties(true));
            }
            else {
                response.content.isProcessing = false;
                return response.content;
            }
        });
    }
    getSpecificProperties(accessToken, urn, modelGuid, specificPropertiesPayload, optionalArgs) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.metadataApi.fetchSpecificProperties(accessToken, urn, modelGuid, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.acceptEncoding, (_a = optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.region) !== null && _a !== void 0 ? _a : model_1.Region.Us, specificPropertiesPayload, optionalArgs === null || optionalArgs === void 0 ? void 0 : optionalArgs.options);
            if (response.response.status == 202) {
                return (new model_1.SpecificProperties(true));
            }
            else {
                response.content.isProcessing = false;
                return response.content;
            }
        });
    }
}
exports.ModelDerivativeClient = ModelDerivativeClient;
