// @aps_sdk/autodesk-sdkmanager/dist/esm/src/SdkManager.js
import { ApsConfiguration } from './ApsConfiguration';
import { ResiliencyConfiguration } from './ResiliencyConfiguration';
import { ApsClient } from './ApsClient';
import { Logger } from './Logger';
export class SdkManager {
    constructor(apsConfiguration, resiliencyConfiguration, authClient, logger) {
        this._apsConfiguration = apsConfiguration !== null && apsConfiguration !== void 0 ? apsConfiguration : new ApsConfiguration({});
        this._resiliencyConfiguration =
            resiliencyConfiguration !== null && resiliencyConfiguration !== void 0 ? resiliencyConfiguration : new ResiliencyConfiguration();
        this._apsClient = new ApsClient(this._resiliencyConfiguration);
        this._authClient = authClient;
        this._logger = logger !== null && logger !== void 0 ? logger : new Logger();
    }
    set logger(logger) {
        this._logger = logger;
    }
    get logger() {
        return this._logger;
    }
    static getInstance(apsConfiguration, resiliencyConfiguration, authClient, logger) {
        return new SdkManager(apsConfiguration, resiliencyConfiguration, authClient, logger);
    }
    get apsClient() {
        return this._apsClient;
    }
    get authClient() {
        return this._authClient;
    }
    get apsConfiguration() {
        return this._apsConfiguration;
    }
    set apsConfiguration(apsConfiguration) {
        this._apsConfiguration = apsConfiguration;
    }
}
